docs/content/sips/draft-000-templates.md

```diff
@@ -0,0 +1,365 @@
+title = "SIP DRAFT-000 - Templates"
+template = "main"
+date = "2022-03-22T14:53:30Z"
+
+---
+
+Summary: Proposes changes to the `spin new` syntax and to the template
+authoring format
+
+Created: March 28, 2022
+
+Status: DRAFT
+
+## Background
+
+We have an initial cut of a template system planned for 0.1, but design and UX
+work on this was somewhat constrained by the 0.1 deadline.  The 0.1 design is
+closely based on `cargo generate` because it was a readily available, er,
+template in the language we were using.  This proposal discusses goals and
+design options for taking the template system forward.
+
+The existing work gives us a solid basis with a well-documented template syntax
+using the Liquid engine, safe failure modes via staging, and other good
+stuff.  No changes are proposed to that core.  This SIP relates to user
+experience (for both consumers and authors) and broader architecture.
+
+## Goals
+
+* Convenient, forgiving user experience for consumers
+* Foundation for delivery via IDE
+* Minimise what authors need to know
+
+## Should we have a template system at all?
+
+It's worth considering this because there are a ton of existing templating
+systems, some language specific (e.g. `cargo generate`, `dotnet new`), some
+generic (e.g. Yeoman).  These are likely to be idiomatic to the languages
+people use.  Is it worth having our own?
+
+Reasons why we should:
+
+* Some languages _don't_ have template support, or require the user to go
+  install yet more tools.
+* It allows documentation to assume Spin templates instead of having to
+  write different docs for every language.
+* Having our own system doesn't preclude offering Spin templates via
+  language-idiomatic systems too.
+
+Reasons why we shouldn't:
+
+* It's a decent chunk of work to do it well.
+* It's extra code for every copy of Spin to carry around.
```

Comment:
Considering our early discussions around splitting the runtime part of Spin, and distributing the templates implementation only for developers, would this still be a concern?